### DISCUSSION TITLE
Drop and Create SQL Server database

### DISCUSSION THREAD

**Discussion by Obish** [lecaillon/Evolve#206] [Q&A]
Hi

i want to confirm if Evolve can drop or create a database in sql server

**Comment by lecaillon** [comment]
Yes, even if some reported issues I do not reproduce.

In the SQL Server integration tests of Evolve, you can see a database is created [here](https://github.com/lecaillon/Evolve/blob/master/test/Evolve.Tests/Integration/SQLServer/Resources/Sql_Scripts/Migration/V3_1__create_database_evolvedb.sql), then a table in the new database [here](https://github.com/lecaillon/Evolve/blob/master/test/Evolve.Tests/Integration/SQLServer/Resources/Sql_Scripts/Migration/V3_1_1__create_table_evolve.sql)

Just use the special comment `-- evolve-tx-off` in the first line of your file, when you execute SQL Server commands that do not support transactions

**Reply by Obish** [reply]
Thank you for the response.


So what DB name should I specify in the connection string?

The one i'm about to create?

**Reply by lecaillon** [reply]
Do not add the database in your connection string or put master. Then in your scripts use the full name to identify your objects like in the scripts above. 

By the way, the "standard" use of Evolve, don't recommend database creation scripts. Juste schema. But it is possible.

**Comment by lecaillon** [comment]
More information there : https://github.com/lecaillon/Evolve/issues/97#issuecomment-792609548

**Comment by pbparentpay** [comment]
> Hi
> 
> i want to confirm if Evolve can drop or create a database in sql server

Has anyone tried to create the database? What needs to be connectionString? I am setting `connectionString="Server=xxxx;Database=evolvedb;User Id=xxx;Password=xxxx;"` and create database sql file as [https://github.com/lecaillon/Evolve/blob/master/test/Evolve.Tests/Integration/SQLServer/Resources/Sql_Scripts/Migration/V3_1__create_database_evolvedb.sql](url)
But I am getting error. 
`Evolve.EvolveException: 'Validation of the database connection failed. Cannot open database "evolvedb" requested by the login. The login failed.`
But If I create database first in ms sql server manually, and just run create tables sql using Evolve, it's working. 
What am I missing?
Platform: .Net Core 6
Db: MS SQL Server
